			     BASH PATCH REPORT
			     =================

Bash-Release:	4.4
Patch-ID:	bash44-017

Bug-Reported-by:	ZhangXiao <xiao.zhang@windriver.com>
Bug-Reference-ID:	<58AD3EAC.4020608@windriver.com>
Bug-Reference-URL:	http://lists.gnu.org/archive/html/bug-bash/2017-02/msg00061.html

Bug-Description:

There is a memory leak when `read -e' is used to read a line using readline.

Patch (apply with `patch -p0'):

*** ../bash-20170217/builtins/read.def	2017-01-02 16:53:02.000000000 -0500
--- builtins/read.def	2017-02-22 09:43:14.000000000 -0500
***************
*** 691,694 ****
--- 691,699 ----
    CHECK_ALRM;
  
+ #if defined (READLINE)
+   if (edit)
+     free (rlbuf);
+ #endif
+ 
    if (retval < 0)
      {
*** ../bash-4.4/patchlevel.h	2016-06-22 14:51:03.000000000 -0400
--- patchlevel.h	2016-10-01 11:01:28.000000000 -0400
***************
*** 26,30 ****
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 16
  
  #endif /* _PATCHLEVEL_H_ */
--- 26,30 ----
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 17
  
  #endif /* _PATCHLEVEL_H_ */
